criu/uffd.c

```diff
@@ -313,9 +314,11 @@ int setup_uffd(int pid, struct task_restore_args *task_args)
 	 * Open userfaulfd FD which is passed to the restorer blob and
 	 * to a second process handling the userfaultfd page faults.
 	 */
-	task_args->uffd = uffd_open(O_CLOEXEC | O_NONBLOCK, &features);
+	task_args->uffd = uffd_open(O_CLOEXEC | O_NONBLOCK, &features, false);
 	if (task_args->uffd < 0) {
-		pr_perror("Unable to open an userfaultfd descriptor");
+		if (task_args->uffd == -errno)
```

Comment:
I'm not sure I understand ths, why compare to `-errno`?

Reply:
In any case, I think what rppt suggested in https://github.com/checkpoint-restore/criu/pull/1094#pullrequestreview-428418065 is this

There are only 2 callers to uffd_open. Instead of adding an additional parameter to control the log level there, their suggestion is to move the logging responsibility to the caller itself.

Reply:
I see. I also noticed I made a critical error by not providing a different return value if log_level is true. My intent was to have uffd_open() print info if log_level was set to true, else return -errno so the caller can print the error. However, I did not execute this properly. 

I think you are right that I should have the callers be responsible for printing the error/info.

Reply:
I use -errno because there were different areas in uffd_open where something could have gone wrong. To make the message more specific, I wanted -errno to indicate that there were no available lazy pages. My intent with using -errno came from rppt's suggestion of changing the wording of the errors. 

However, I think you are right. I will make the callers responsible for handling whether to print as info or error. Thanks, @kashyapp !